src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -1,462 +1,407 @@
 
-    [Fact]
-    public static void CreateDirectory_FileWithoutTrailingSlashAsPath_ThrowsIOException()
-    {   // VSWhidbey #104049
-        using (TemporaryFile file = new TemporaryFile())
+            Assert.Equal(path.FullPath, result.FullName);
+            Assert.True(Directory.Exists(result.FullName));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void DirectoryWithComponentLongerThanMaxComponentAsPath_ThrowsPathTooLongException()
         {
-            string path = IOServices.RemoveTrailingSlash(file.Path);
+            // While paths themselves can be up to 260 characters including trailing null, file systems
+            // limit each components of the path to a total of 255 characters.
+            var paths = IOInputs.GetPathsWithComponentLongerThanMaxComponent();
 
-            Assert.Throws<IOException>(() =>
+            Assert.All(paths, (path) =>
             {
-                Directory.CreateDirectory(path);
+                Assert.Throws<PathTooLongException>(() => Create(path));
             });
         }
-    }
-
 
-    [Fact]
-    public static void CreateDirectory_FileWithTrailingSlashAsPath_ThrowsIOException()
-    {
-        using (TemporaryFile file = new TemporaryFile())
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void DirectoryLongerThanMaxPathAsPath_ThrowsPathTooLongException()
         {
-            string path = IOServices.AddTrailingSlashIfNeeded(file.Path);
-
-            Assert.Throws<IOException>(() =>
+            var paths = IOInputs.GetPathsLongerThanMaxPath();
+            Assert.All(paths, (path) =>
             {
-                Directory.CreateDirectory(path);
+                Assert.Throws<PathTooLongException>(() => Create(path));
             });
         }
-    }
 
-    [Fact]
-    public static void CreateDirectory_ExistingDirectoryWithoutTrailingSlashAsPath_DoesNotThrow()
-    {
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void DirectoryLongerThanMaxDirectoryAsPath_ThrowsPathTooLongException()
         {
-            string path = IOServices.RemoveTrailingSlash(directory.Path);
-
-            DirectoryInfo result = Directory.CreateDirectory(path);
-
-            Assert.Equal(path, result.FullName);
-            Assert.True(Directory.Exists(result.FullName));
+            var paths = IOInputs.GetPathsLongerThanMaxDirectory();
+            Assert.All(paths, (path) =>
+            {
+                Assert.Throws<PathTooLongException>(() => Create(path));
+            });
         }
-    }
 
-    [Fact]
-    public static void CreateDirectory_ExistingDirectoryWithTrailingSlashAsPath_DoesNotThrow()
-    {
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        [ActiveIssue(645)]
```

Comment:
The way this is written, this isn't going to run on any platform, but it's fine on Unix, right?  I think what you meant was:
```C#
[Fact]
[ActiveIssue(645, PlatformID.Windows)]
public void LongPath_Allow() { ... }
```
which says "run this test on every platform except for Windows because there's an active issue preventing this from working on Windows".